resources/Login.py

```diff
@@ -0,0 +1,36 @@
+from werkzeug.security import check_password_hash
+from flask_restful import Resource
+from flask import request
+from middleware.login_queries import login_results
+
+
+class Login(Resource):
+    def __init__(self, **kwargs):
+        self.psycopg2_connection = kwargs["psycopg2_connection"]
+
+    def post(self):
+        """
+        Login function: allows a user to login using their email and password as credentials
+        The password is compared to the hashed password stored in the users table
+        Once the password is verified, an API key is generated, which is stored in the users table and sent to the verified user
+        """
+        try:
+            data = request.get_json()
+            email = data.get("email")
+            password = data.get("password")
+            cursor = self.psycopg2_connection.cursor()
+
+            user_data = login_results(cursor, email)
+
+            if "password_digest" in user_data and check_password_hash(
+                user_data["password_digest"], password
+            ):
+                return {
+                    "message": "Successfully logged in",
+                    "data": user_data["api_key"],
```

Comment:
Hey @mbodeantor Quick question here: Is this API key functioning as an access token for authenticating sessions or is it a unique key per user that is stable for accessing the API?
I don't think we want to use stable API keys as access tokens. Could we use a JWT instead? Could very easily encode an expiry time into the token so that we can use a refresh token to generate a new access token when a user's access token is about to expire and they're still using the app. I believe this is the strategy we discussed previously with @josh-chamberlain, no?

Reply:
should be fixed now